doc/source/gallery/geopandas_rasterio_sample.ipynb

```diff
@@ -0,0 +1,303 @@
+   "outputs": [],
+   "source": [
+    "import geopandas\n",
+    "import rasterio\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Load the vector data\n",
+    "=============================\n",
+    "\n",
+    "Let's open a geopackage of the points we want to use to sample the raster data\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "gdf = geopandas.read_file('data\\\\sample_points.gpkg')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The ``GeoDataFrame`` looks like this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
+       "  <thead>\n",
+       "    <tr style=\"text-align: right;\">\n",
+       "      <th></th>\n",
+       "      <th>id</th>\n",
+       "      <th>geometry</th>\n",
+       "    </tr>\n",
+       "  </thead>\n",
+       "  <tbody>\n",
+       "    <tr>\n",
+       "      <th>0</th>\n",
+       "      <td>1</td>\n",
+       "      <td>POINT (615858.1930048079 5619641.60641626)</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>1</th>\n",
+       "      <td>2</td>\n",
+       "      <td>POINT (621640.7523066753 5613571.516541371)</td>\n",
+       "    </tr>\n",
+       "    <tr>\n",
+       "      <th>2</th>\n",
+       "      <td>3</td>\n",
+       "      <td>POINT (622854.7702816529 5605424.816972442)</td>\n",
+       "    </tr>\n",
+       "  </tbody>\n",
+       "</table>\n",
+       "</div>"
+      ],
+      "text/plain": [
+       "   id                                     geometry\n",
+       "0   1   POINT (615858.1930048079 5619641.60641626)\n",
+       "1   2  POINT (621640.7523066753 5613571.516541371)\n",
+       "2   3  POINT (622854.7702816529 5605424.816972442)"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "gdf.head()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Open the raster data\n",
+    "=============================\n",
+    "\n",
+    "Use ``rasterio`` to open the raster data to be sampled"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "src = rasterio.open('data\\\\s2a_l2a_iow.tif')"
```

Comment:
Same as above. We can in theory store some (small) dataset for the purpose of this example or load it over web (from safe source).

Reply:
Here I've added a subset of the original image to keep the filesize down.  I hope this is of an appropriate size.